ProfileManager.js

```diff
@@ -938,15 +958,18 @@ export default class ProfileManager {
   async _getAgentSigner({profileAgentId, useEphemeralSigner}) {
     const cacheKey = `${profileAgentId}-${useEphemeralSigner}`;
     const agentSignersCache = this._getCache('agent-signers');
-    const agentSigner = agentSignersCache.get(cacheKey);
+    let agentSigner = agentSignersCache.get(cacheKey);
     if(agentSigner) {
       return agentSigner;
     }
 
     const promise = this._createAgentSigner(
       {profileAgentId, useEphemeralSigner});
-
-    agentSignersCache.set(cacheKey, promise);
+    agentSigner = await promise;
+    const now = Date.now();
+    const expiryDate = new Date(agentSigner.expires);
```

Comment:
`agentSigner` isn't a zcap, right? So it wouldn't have `expires`. We have to see if `agentSigner.capability` has `expires`, and if so, use its `expires` property. If not, we can default to `maxAge = this.zcapTtl - this.zcapGracePeriod`.